src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -1947,8 +1947,59 @@ void c_typecheck_baset::typecheck_side_effect_function_call(
     if(symbol_table.symbols.find(identifier)==symbol_table.symbols.end())
     {
       // This is an undeclared function.
+
+      // Is it the polymorphic typed_target function ?
+      if(identifier == CPROVER_PREFIX "typed_target")
+      {
+        if(expr.arguments().size() != 1)
+        {
```

Comment:
I would appreciate if the body of this `if` statement could be factored out into a function to reduce the overall size (in the number of lines) of the encompassing function.

Reply:
fixed